ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -1332,9 +1332,16 @@ module VM = struct
       @ affinity
       @ weight
     in
+    let generation_id =
+      match vm.generation_id with
+      | Some genid ->
+          [("generation-id", genid)]
+      | None ->
```

Comment:
Will this handle migration correctly

Reply:
Migration also includes the VM metadata (`vm`). And if this metadata does not include a `generation_id`, in case it came from an older version, then the key will still be in the platformdata. But I'd better do something to ensure that there can't be two versions of the key in platformdata, just in case.